x/datapool/keeper/pool.go

```diff
@@ -436,6 +454,19 @@ func (k Keeper) increaseCurNumAndUpdatePool(ctx sdk.Context, pool *types.Pool) {
 	k.SetPool(ctx, pool)
 }
 
+func (k Keeper) mintPoolShareToken(ctx sdk.Context, poolID, amount uint64) error {
+	k.setInitialSupply(ctx, poolID)
```

Comment:
How about we separate this `setInitialSupply()` out from `mintPoolShareToken()`? 
If next round goes, data pool may mint more `ShareTokens` as much as increased `targetNumData`. No more initialization of supply may be required in next round curation.

Reply:
Oh good. I'll fixed it